### PR TITLE
Add inline docs to all headers, restructure the application

### PR DESCRIPTION
**Pull request by janjongboom**
@MarceloSalazar @screamerbg

### CODE REVIEW

**Comment by screamerbg** [comment]
This is awesome! Thanks @janjongboom!

**Comment by screamerbg** [comment]
@janjongboom The inline documentation is great. Frankly I find the lots of folders very confusing, e.g. header files in one folder and sources in another, instead of living side by side. I know that some of PDMC is structured this way, but it's very counter intuitive.

As an example, having `simple-mbed-cloud-client/simple-mbed-cloud-client.h` inside a repo `simple-mbed-cloud-client` (and it's a single file) feels wrong. 

Any chance to have the original file structure? or may be with a few folders, but not this yotta-like structure.

File: simple-mbed-cloud-client/simple-mbed-cloud-client.h [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+     * @returns true if connection started, false if connection was not started
+     */
+    bool call_register();
+
+    /**
+     * Create MbedCloudClientResource on the underlying Mbed Cloud Client object,
+     * register with Pelion Device Management.
+     *
+     * This function can only be called once. If you need to re-register,
+     * use `call_register`.
```
Comment (MarceloSalazar) [inline simple-mbed-cloud-client/simple-mbed-cloud-client.h]: I think this behavior should be improved, so there is no risk of calling the function twice.

**Comment by MarceloSalazar** [comment]
I agree with some of @screamerbg comments on the folders.
I'd like to have at least content from source and simple-mbed-cloud-client in the same folder.

**Comment by screamerbg** [comment]
All tests are passing with this PR:

```
+-------------------------+---------------------+-------------------------------------------------+---------+---------+
| target                  | platform_name       | test suite                                      | result  |    time |
+-------------------------+---------------------+-------------------------------------------------+---------+---------+
| NUMAKER_PFM_M487-ARM    | NUMAKER_PFM_M487    | simple-mbed-cloud-client-tests-dev_mgmt-connect | OK      |   64.49 |
| NUMAKER_PFM_M487-ARM    | NUMAKER_PFM_M487    | simple-mbed-cloud-client-tests-dev_mgmt-update  | OK      |  136.33 |
| K66F-ARM                | K66F                | simple-mbed-cloud-client-tests-dev_mgmt-connect | OK      |   82.81 |
| K66F-ARM                | K66F                | simple-mbed-cloud-client-tests-dev_mgmt-update  | OK      |  167.94 |
| NUCLEO_F429ZI-ARM       | NUCLEO_F429ZI       | simple-mbed-cloud-client-tests-dev_mgmt-connect | OK      |   73.61 |
| NUCLEO_F429ZI-ARM       | NUCLEO_F429ZI       | simple-mbed-cloud-client-tests-dev_mgmt-update  | OK      |  164.41 |
| MTB_USI_WM_BN_BM_22-ARM | MTB_USI_WM_BN_BM_22 | simple-mbed-cloud-client-tests-dev_mgmt-connect | OK      |  111.36 |
| MTB_USI_WM_BN_BM_22-ARM | MTB_USI_WM_BN_BM_22 | simple-mbed-cloud-client-tests-dev_mgmt-update  | OK      |  276.81 |
| NUCLEO_F767ZI-ARM       | NUCLEO_F767ZI       | simple-mbed-cloud-client-tests-dev_mgmt-connect | OK      |   64.55 |
| NUCLEO_F767ZI-ARM       | NUCLEO_F767ZI       | simple-mbed-cloud-client-tests-dev_mgmt-update  | OK      |  146.79 |
| UBLOX_EVK_ODIN_W2-ARM   | UBLOX_EVK_ODIN_W2   | simple-mbed-cloud-client-tests-dev_mgmt-connect | OK      |  157.99 |
| UBLOX_EVK_ODIN_W2-ARM   | UBLOX_EVK_ODIN_W2   | simple-mbed-cloud-client-tests-dev_mgmt-update  | OK      |  549.07 |
| NUCLEO_F746ZG-ARM       | NUCLEO_F746ZG       | simple-mbed-cloud-client-tests-dev_mgmt-connect | OK      |   69.22 |
| NUCLEO_F746ZG-ARM       | NUCLEO_F746ZG       | simple-mbed-cloud-client-tests-dev_mgmt-update  | OK      |  152.07 |
| NUMAKER_IOT_M487-ARM    | NUMAKER_IOT_M487    | simple-mbed-cloud-client-tests-dev_mgmt-connect | OK      |   65.14 |
| NUMAKER_IOT_M487-ARM    | NUMAKER_IOT_M487    | simple-mbed-cloud-client-tests-dev_mgmt-update  | OK      |  173.78 |
| K64F-ARM                | K64F                | simple-mbed-cloud-client-tests-dev_mgmt-connect | OK      |   84.99 |
| K64F-ARM                | K64F                | simple-mbed-cloud-client-tests-dev_mgmt-update  | OK      |  179.68 |
| MTB_MXCHIP_EMW3166-ARM  | MTB_MXCHIP_EMW3166  | simple-mbed-cloud-client-tests-dev_mgmt-connect | OK      |   112.5 |
| MTB_MXCHIP_EMW3166-ARM  | MTB_MXCHIP_EMW3166  | simple-mbed-cloud-client-tests-dev_mgmt-update  | OK      |  233.46 |
| NUMAKER_PFM_NUC472-ARM  | NUMAKER_PFM_NUC472  | simple-mbed-cloud-client-tests-dev_mgmt-connect | OK      |  140.27 |
| NUMAKER_PFM_NUC472-ARM  | NUMAKER_PFM_NUC472  | simple-mbed-cloud-client-tests-dev_mgmt-update  | OK      |  219.88 |
+-------------------------+---------------------+-------------------------------------------------+---------+---------+

2018-12-11 17:22:05 TEST FINISHED

**Comment by screamerbg** [comment]
I'm happy with the PR as it is. I'd rather address @MarceloSalazar's comments in a separate PR.

**Pull request merged**